mdtraj/core/topology.py

```diff
@@ -1200,15 +1278,41 @@ def select_atom_indices(self, selection="minimal"):
         if selection == "all":
             atom_indices = np.arange(self.n_atoms)
         elif selection == "alpha":
-            atom_indices = [a.index for a in self.atoms if a.name == "CA" and a.residue.is_protein]
+            atom_indices = np.array(
+                [
+                    a.index
+                    for a in self.atoms
+                    if a.name == "CA" and a.residue.is_protein
+                ],
+                dtype=np.int32,
+            )
         elif selection == "minimal":
-            atom_indices = [
-                a.index for a in self.atoms if a.name in ["CA", "CB", "C", "N", "O"] and a.residue.is_protein
-            ]
+            atom_indices = np.array(
+                [
+                    a.index
+                    for a in self.atoms
+                    if a.name in ["CA", "CB", "C", "N", "O"] and a.residue.is_protein
+                ],
+                dtype=np.int32,
+            )
         elif selection == "heavy":
-            atom_indices = [a.index for a in self.atoms if a.element != elem.hydrogen and a.residue.is_protein]
+            atom_indices = np.array(
+                [
+                    a.index
+                    for a in self.atoms
+                    if a.element != elem.hydrogen and a.residue.is_protein
+                ],
+                dtype=np.int32,
```

Comment:
I'm not sure we should set the dtype like this explicitly. It may conflict with the default integer type on most systems (which is what would have been returned previously)

I'm also thinking of the previous PR we did with NumPy 2.0 where LAMMPS was using 32 bit numbers (causing some issues) and it was unclear why that choice was made. If we make an explicit choice on types like this for a data structure, it should be documented and for a reason rather than seeming somewhat arbitray.

Reply:
I agree. I have changed all the `np.int32` to a more general `np.integer` to be on the safe side.